src/parameterBuilder.js

```diff
@@ -13,26 +13,48 @@ function typeDeclarations( definition ) {
 	} );
 }
 
-function buildArrayParam( key, param ) {
+function addArrayParamExpansion( request, key, param ) {
+	const matcher = new RegExp( `(\\(\\s*@${ key }\\s*\\))|@${ key }\\b`, "ig" );
+	if ( param.val.length === 0 ) {
+		request.sqlTextOrProcedure = request.sqlTextOrProcedure.replace( matcher, "(SELECT 1 WHERE 1=0)" );
+		return;
+	}
+
+	const { type: { type, length, precision, scale } } = param;
+
+	const params = [];
+	let i = 0;
+	for ( const item of param.val ) {
```

Comment:
I can't help trying to make this loop less imperative. I might be reaching too hard here, but what if you set `params` to a `map` over `param.val`? That would save you from declaring and incrementing `i`. Then you could still call `request.addParameter` in the mapped function.

Reply:
Feels wrong to have a map with side effects though.

Reply:
I thought you might say that! And I agree. My only other idea was a reduce with the request and params in a tuple as the accumulator.